test/cpp/api/CMakeLists.txt

```diff
@@ -51,6 +51,7 @@ endif()
 
 add_executable(test_api ${TORCH_API_TEST_SOURCES})
 target_include_directories(test_api PRIVATE ${ATen_CPU_INCLUDE})
+target_link_libraries(test_api PRIVATE torch gmock)
```

Comment:
Interesting that there's no usage of gmock already in the codebase!

Reply:
There's a few, just not in the api tests here.

```
$ git grep gmock | grep --invert-match ^third_party/
aten/src/ATen/core/boxing/impl/test_helpers.h:#include <gmock/gmock.h>
aten/src/ATen/test/Dict_test.cpp:#include <gmock/gmock.h>
aten/src/ATen/test/ivalue_test.cpp:#include <gmock/gmock.h>
aten/src/ATen/test/vitals.cpp:#include <gmock/gmock.h>
c10/test/CMakeLists.txt:    target_link_libraries(${test_name} c10 gmock gtest gtest_main)
c10/test/util/either_test.cpp:#include <gmock/gmock.h>
c10/test/util/optional_test.cpp:#include <gmock/gmock.h>
c10/test/util/string_view_test.cpp:#include <gmock/gmock.h>
cmake/Dependencies.cmake:  set(BUILD_GMOCK ON CACHE BOOL "Build gmock." FORCE)
test/cpp/api/CMakeLists.txt:target_link_libraries(test_api PRIVATE torch gmock)
test/cpp/api/modules.cpp:#include <gmock/gmock.h>
test/cpp/jit/test_interpreter.cpp:#include <gmock/gmock.h>
```